[Merl 491] - Document `getNextServerSideProps` and `getNextStaticProps` (#1076)
wpengine/faustjs: internal/faustjs.org/docs/next/reference/getNextServerSideProps.mdx

---
slug: /next/reference/getNextServerSideProps
title: getNextServerSideProps
description: The getNextServerSideProps function lets you server side render your page with WordPress data outside of the template hierarchy.
---

The `getNextServerSideProps` function lets you server side render your page with WordPress data outside of the template hierarchy using Next.js file based pages.
The function should be returned from [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props) which is required by Next.js to perform Server-Side Rendering (SSR):

```tsx
export async function getServerSideProps(context: GetServerSidePropsContext) {
  return getNextServerSideProps(context, {
    Page: MyPage,
  });
}
```

The `getNextServerSideProps` function accepts two arguments: the server side props `context`, and an object (type `GetNextServerSidePropsConfig`) with a `Page` key.
This should be your Next.js page component.

> The reason `MyPage` are passed to `getNextServerSideProps` under the hood Faust will use the Apollo Client
> to fetch the data using the 'query' and 'variables' properties. If the `Page` does not have those properties then no queries will be performed.

Reviewer: ```suggestion
Faust uses `MyPage` passed via `getNextServerSideProps` to fetch the data using the 'query' and 'variables' properties. If the `Page` does not have those properties then no queries will be performed.
```

> This is merely a convenient way to fetch the page data while reusing the logic between components. If you are interested on how this is done you can review the [code section here](https://github.com/wpengine/faustjs/blob/canary/packages/faustwp-core/src/getProps.ts#L71)

Reviewer: ```suggestion
This is merely a convenient way to fetch the page data while reusing the logic between components. If you are interested on how this is done you can review the [source code here.](https://github.com/wpengine/faustjs/blob/canary/packages/faustwp-core/src/getProps.ts#L71)
```
